Create our PR lifecycle document (#50)
anitab-org/documentation: PULL_REQUEST_LIFECYCLE.md

# Pull request lifecycle

* PR Template is necessary
* PR needs to link to the issue #
    * Issue needs to be validated (at least 1 person from the “core team” of the project)
    * Issue needs to be assigned to the person who submitted the PR 

**Status: Needs Review**

> * Minimum Reviews = 2 PR Reviews from any person (at least 1 PR Review from a Maintainer)

Reviewer: Hi @isabelcosta I have a doubt here. May be it is should be:-
```suggestion
* Minimum Reviews = 2 PR Reviews from any contributor (at least 1 PR Review from a Maintainer)
```

* if successful, 

>     * Status Change → Label “Status: Needs Testing”

Author: ```suggestion
    * Status Change → Label “Status: Needs Testing”
        * if the PR is a documentation PR, we can label it “Status: Ready to Merge”
```

> * if not successful, the Reviewer pings the author to let them know where PR failed. Contributor is responsible for making changes.

Reviewer: Is it possible to clear what is meant by PR failed?
```suggestion
* if not successful, the Reviewer pings the author to let them know various scopes for improvement. Contributor is responsible for making changes.
```

# Pull request lifecycle

* PR Template is necessary
* PR needs to link to the issue #
    * Issue needs to be validated (at least 1 person from the “core team” of the project)

>     * Issue needs to be assigned to the person who submitted the PR 
> 

Author: ```suggestion
    * Issue needs to be assigned to the person who submitted the PR 
    * **Exception:** this may not be needed if the PR is about small typo fixes
```

**Status: Needs Review**

* Minimum Reviews = 2 PR Reviews from any contributor (at least 1 PR Review from a Maintainer)
* if successful, 
    * Status Change → Label “Status: Needs Testing”
        * if the PR is a documentation PR, we can label it “Status: Ready to Merge”

> * if not successful, the Reviewer pings the author to let them know various scopes for improvement. Contributor is responsible for making changes.

Reviewer 2: ```suggestion
* if not successful, the Reviewer pings the contributor to let them know various scopes for improvement. Contributor is responsible for making changes.
```

    * Status Change → Label “Status: Changes Requested”

**Status: Needs Testing**

> * Minimum QA = 1 QA report (using [this template](quality-assurance.md#how-to-test-a-pr)) from any person (except the author)

Reviewer 2: ```suggestion
* Minimum QA = 1 QA report (using [this template](quality-assurance.md#how-to-test-a-pr)) from any person (except the contributor)
```

* If successful, 
    * Status Change → Label “Status: Ready to Merge”

> * If not successful, QA pings author to let them know where PR failed. Contributor is responsible for making changes.

Reviewer 2: ```suggestion
* If not successful, QA pings the contributor to let them know where PR failed. Contributor is responsible for making changes.
```
